### PR TITLE
Fix bugs for user input validation in add command, extract method for junit tests

### LINKED ISSUE #127: Able to add empty description entry
image
image

### LINKED ISSUE #131: Logic error with checking for user input leading to wrong exception msg
image

### PR DESCRIPTION
**Pull request by wx-03**
- Change logic for add command user input validation. If user input is `add spending randomPrice macs` it will show amount missing instead of "amount is not a number" because randomPrice could have been part of the description.
- Add method `commandInputForTest()` in Ui class to parse and execute user input, but I'm not sure if Ui is the correct class to put it in. 

Closes #127 
Closes #131

### CODE REVIEW

File: src/test/java/seedu/commands/AddCommandTest.java [hunk 1]
```diff
@@ -46,164 +46,149 @@ public void restore() {
...
 
     @Test
-    void addCommand_invalidCategorySpendingInput_expectIllegalInputExceptionThrown() {
-        String userInput = "add invalidCategory 1 s";
-        Command c = Parser.parseUserInput(userInput);
-        c.execute(incomes, spendings);
+    void execute_invalidCategorySpendingInput_expectIllegalInputExceptionThrown() {
+        commandInputForTest("add invalidCategory 1 s", incomes, spendings);
         assertEquals(TAB + INVALID_CATEGORY + ADD_COMMAND_FORMAT
                         + System.lineSeparator(), outContent.toString());
```
Comment (NigelYeoTW) [inline src/test/java/seedu/commands/AddCommandTest.java:92]: 16 spaces seems to be incorrect

**Review by NigelYeoTW — COMMENTED** [review]
Tested the add command, seems to be working really well, nice. There seem to be some inconsistency with the next line for test as highlighted by my comment could help check and rectify the coding standard as well for those, ty

File: src/test/java/seedu/commands/AddCommandTest.java (hunk 1 above)
Comment (wx-03) [reply src/test/java/seedu/commands/AddCommandTest.java]: Fixed

**Comment by NigelYeoTW** [comment]
lgtm

**Pull request merged**